core/include/irods/private/http_api/openid.hpp

```diff
@@ -0,0 +1,24 @@
+#ifndef IRODS_HTTP_API_OPENID_HPP
+#define IRODS_HTTP_API_OPENID_HPP
+
+#include <optional>
+#include <string>
+
+#include <jwt-cpp/jwt.h>
+#include <jwt-cpp/traits/nlohmann-json/traits.h>
+#include <nlohmann/json.hpp>
+
+#include <boost/beast/http/message.hpp>
+#include <boost/beast/http/string_body.hpp>
+#include <boost/url/url_view.hpp>
+
+namespace irods::http::openid
+{
+	auto create_oidc_request(boost::urls::url_view _url)
+		-> boost::beast::http::request<boost::beast::http::string_body>;
+	auto validate_using_introspection_endpoint(const std::string& _bearer_token) -> std::optional<nlohmann::json>;
+	auto validate_using_local_validation(const jwt::decoded_jwt<jwt::traits::nlohmann_json>& _jwt)
+		-> std::optional<nlohmann::json>;
```

Comment:
Add empty line between function declarations. Just makes it a little easier to read.

Reply:
Should I also throw the doxygen up there?

Reply:
Done.

Reply:
> Should I also throw the doxygen up there?

Sure. That will only help us in the future.